### ISSUE TITLE
proper treatment of error messages

### ISSUE THREAD

**Issue by test-fullautomation** [test-fullautomation/robotframework-testsuitesmanagement#114]
I wonder about the output here.

As I understand we result in the **test result state "unknown"** because we find an error (Could not set variable...). This is no "unkown exception".

I would expect the following:
```Error: Could not set variable 'params['global']['teststring_bench']' with value 'I am the 'bench1' configuration of exercise 08'!```
then:
`BuiltIn().unknown()`

No traceback, test-result "unknown".

General:
A traceback indicates for me a severe unhandled issue in the testsuites-management or robotframework core (or anywhere else).
Our implementations should always consider all possible error scenarios that it never comes to a traceback.
A handled error should never end with a traceback.

```
UNKNOWN - Unknown exception occurs. Details: Could not set variable 'params['global']['teststring_bench']' with value 'I am the 'bench1' configuration of exercise 08'!
DEBUG - Traceback (most recent call last):
  File "D:\RobotFramework\python39\lib\site-packages\RobotFramework_TestsuitesManagement\__init__.py", line 67, in run_keyword
    return DynamicCore.run_keyword(self, name, args, kwargs)
  File "D:\RobotFramework\python39\lib\site-packages\robotlibcore.py", line 105, in run_keyword
    return self.keywords[name](*args, **(kwargs or {}))
  File "D:\RobotFramework\python39\lib\site-packages\RobotFramework_TestsuitesManagement\Keywords\CSetup.py", line 70, in testsuite_setup
    BuiltIn().unknown(CConfig.sLoadedCfgError)
  File "D:\RobotFramework\python39\lib\site-packages\robotframework-4.1.3-py3.9.egg\robot\libraries\BuiltIn.py", line 562, in unknown
    raise UnknownAssertionError(msg) if msg else UnknownAssertionError()
```

**Comment by namsonx** [comment]
Hello Thomas,

In case we want to remove traceback for all Robot keywords which we define in testsuitesmanagement library, I will comment out overwritten method "def run_keyword(self, name, args, kwargs):" in __init__.py of testsuitesmanagement package.
Otherwise, If you want to remove traceback log for unknown() case only, Cuong has to modify code in Robot Framework core.

Thank you,
Son

**Comment by test-fullautomation** [comment]
Hi Son,

Generally:
we need tracebacks if we really have a crash.
But we should whenever possible consider possible crashs in our code and handle them. If we handle this, then we don't need a traceback, but a speaking error message.

In this case we have to take care that `DynamicCore.run_keyword(self, name, args, kwargs)` raises no exception by handling all possible exceptions in our keywords.

So I think Cuong has to avoid the traceback in case of  `unknown` state.

Thank you,
 Thomas

**Comment by namsonx** [comment]
Hello Thomas,

Yes, I already discussed with Cuong on this ticket, and he said he will modify in Robot FW core to avoid this traceback.

Thank you,
Son

**Comment by test-fullautomation** [comment]
very good!

**Comment by namsonx** [comment]
Hello @milanac030988 ,

Could you please update the status for this ticket?

Thank you,
Son

**Comment by milanac030988** [comment]
Hello @test-fullautomation 

I have reviewed your expectation in the meeting regarding setting unknown status for import errors and found the following:

- Current status: The current mechanism of Robotframework is that when there is an error in the import section(Library, Resource, Variables), it will print an error log line '[ERROR]...' but still continue executing the test file as usual.

- By changing it, I want you to confirm that you would like to change its current behavior and treat it as a normal Unknown exception (raise unknown and skip executing the test).

- I also have a question to ask for your opinion, whether you want to treat DataErrors as an Unknown exception. As far as I can see, all DataErrors (listed in the attachment) can be reasonably treated as Unknown.

Please let me know your opinion.

[DataErrorList.txt](https://github.com/test-fullautomation/robotframework-testsuitesmanagement/files/10989408/DataErrorList.txt)

**Comment by test-fullautomation** [comment]
Hi Cuong,

I assume that if somebody imports a file, that this has a meaning.
So, if we don't stop execution with unknown, then during test-run an error will occur due to missing resource or keyword.

Therefore "unknown" and skipping executing the test is meaningful for me in this case.

Yes, as far as I see DataErrorList.txt is quite similar to where we use "unknown" in TML.
So, make them all "unknown".

Thank you,
 Thomas

**Comment by test-fullautomation** [comment]
solved and merged with [#18](https://github.com/test-fullautomation/robotframework/pull/18)
